File: C++/Grind75/Week-1/1.two-sum.cpp
#include <bits/stdc++.h>
using namespace std;
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> mp;
        for (int i = 0; i < nums.size(); i++)
        {
            if (mp.find(target - nums[i]) != mp.end())
            {
                return {mp[target - nums[i]], i};
            }
            mp[nums[i]] = i;
        }
        return {0, 0};
    }
};
// @lc code=end

File: Java/Grind75/Week1/1.two-sum.java
import java.util.*;
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (mp.containsKey(target - nums[i])) {
                int ans[] = { mp.get(target - nums[i]), i };
                return ans;
            }
            mp.put(nums[i], i);
        }
        return null;
    }
}
// @lc code=end
